### PR TITLE
Remove `IteratorExt` and `StreamExt`

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

See the linked issue

## 🚫 Blocked by

- #1048 

## 🔍 What does this change?

- Remove `IteratorExt` and `StreamExt`
- Deprecate the `futures` feature

## 📜 Does this require a change to the docs?

The documentation is updated implicitly by removing the `cfg` flags

## 🐾 Next steps

- Re-enable `--feature-powerset` for tests

## 🛡 What tests cover this?

The test snapshots were updated

### CODE REVIEW

File: packages/libs/error-stack/build.rs [hunk 1]
```diff
@@ -18,9 +18,15 @@ fn main() {
         println!("cargo:rustc-cfg=rust_1_65");
     }
 
+    #[cfg(feature = "futures")]
+    println!(
+        "cargo:warning=The `futures` feature for `error-stack` is deprecated as it's not required \
+         anymore"
+    );
+
```
Comment (indietyp) [inline packages/libs/error-stack/build.rs:26]: Maybe we should mention a version where we're going to remove `hooks` and `futures`?

File: packages/libs/error-stack/src/lib.rs [hunk 2]
```diff
@@ -430,8 +425,8 @@
 //! [`set_display_hook`]: Report::set_display_hook
 //! [`set_debug_hook`]: Report::set_debug_hook
 //!
-//! [`Error`]: std::error::Error
-//! [`Error::provide`]: std::error::Error::provide
+//! [`Error`]: core::error::Error
+//! [`Error::provide`]: core::error::Error::provide
```
Comment (indietyp) [inline packages/libs/error-stack/src/lib.rs:429]: Maybe as a drive-by, you could also rename these in all other places (`anyhow`, `fmt`, `eyre`, `compat`, ...)

**Review by indietyp — COMMENTED** [review]
just have some nit-picks

File: packages/libs/error-stack/build.rs (hunk 1 above)
Comment (TimDiekmann) [reply packages/libs/error-stack/build.rs]: I added this in 0a8b49546279404d8af553fd7671bbef06f0236c

**Review by indietyp — APPROVED** [review]

File: packages/libs/error-stack/src/lib.rs [hunk 3]
```diff
@@ -402,25 +402,20 @@
...
-//! `spantrace`| Enables the capturing of [`SpanTrace`]s                        |         | disabled
-//!  `futures` | Provides a [`FutureExt`] adaptor                               |         | disabled
-//!  `anyhow`  | Provides conversion from [`anyhow::Error`] to [`Report`]       |         | disabled
-//!   `eyre`   | Provides conversion from [`eyre::Report`] to [`Report`]        |         | disabled
-//! `pretty-print` | Provide color[^color] and use of unicode in [`Debug`] output |     | enabled
+//!  Feature       | Description                                                        | default
+//! ---------------|--------------------------------------------------------------------|----------
+//! `std`          | Enables support for [`Error`] and, on nightly, [`Backtrace`]       | enabled
+//! `pretty-print` | Provide color[^color] and use of unicode in [`Debug`] output       | enabled
+//! `spantrace`    | Enables the capturing of [`SpanTrace`]s                            | disabled
```
Comment (Alfred-Mountfield) [inline packages/libs/error-stack/src/lib.rs:416]: Automatic capturing?

File: packages/libs/error-stack/src/lib.rs [hunk 4]
```diff
@@ -402,25 +402,20 @@
...
+//! `anyhow`       | Provides conversion from [`anyhow::Error`] to [`Report`]           | disabled
```
Comment (Alfred-Mountfield) [inline packages/libs/error-stack/src/lib.rs:417]: Should we say provides conversion _through_ some mechanism? Like is it through `into_report` or some `From` implementation or?

**Review by Alfred-Mountfield — COMMENTED** [review]
It's a shame to see it go but I do think it ended up being quite confusing, and just increased the footprint of the library anyway.

**Comment by indietyp** [comment]
A side-effect I also found super confusing was that `IteratorExt` would work for `Result<(), core::error::Error>`, but you wanted `ResultExt`.

So instead of telling you there's no way, it would get you to use `IteratorExt`, which (in that case) would be the wrong trait.

File: packages/libs/error-stack/src/lib.rs (hunk 4 above)
Comment (indietyp) [reply packages/libs/error-stack/src/lib.rs:417]: The trait `IntoReportCompat`, which provides `into_report()`, is implemented for `Result<T, eyre::Report>` and `Result<T, anyhow::Error>`.

Comment (Alfred-Mountfield) [reply packages/libs/error-stack/src/lib.rs:417]: I think we should update the description then to say

> `Provides `into_report` to convert [`anyhow::Error`] to [`Report`]

(I'd suggest a change but there are two lines and I only commented on one 🙃)

Comment (TimDiekmann) [reply packages/libs/error-stack/src/lib.rs]: Applied suggestions in c845231da303469eb52123636225aba610ec3b06, thank you!

**Review by Alfred-Mountfield — APPROVED** [review]
Even though I'm sad we're removing a feature we thought would be convenient... I do like the addition/deletion ratio 
image


🚀
